src/component-library/NumberInput/NumberInput.tsx

```diff
@@ -1,39 +1,58 @@
 import * as React from 'react';
+
 import { BaseNumberInput } from './NumberInput.style';
-import { InputProps } from 'component-library/Input';
+import { useNumberField } from '@react-aria/numberfield';
+import { useNumberFieldState } from '@react-stately/numberfield';
+
+type Props = { value?: number; defaultValue?: number; step?: number; label?: string; description?: string };
+
+const defaultProps = {};
 
-// `onWheel` prop can't be used with `preventDefault` because
-// React implements passive event listeners.
-const disableChangeOnWheel = (event: MouseEvent) => {
-  event.preventDefault();
-};
+type NativeAttrs = Omit<React.InputHTMLAttributes<Element>, keyof Props>;
 
-type NumberInputProps = InputProps;
+type NumberInputProps = Props & NativeAttrs & Partial<typeof defaultProps>;
 
 const NumberInput = React.forwardRef<HTMLInputElement, NumberInputProps>(
   (props, ref): JSX.Element => {
-    const inputParent = React.useRef<HTMLDivElement | null>(null);
+    const { disabled, readOnly, required, onChange, ...rest } = props;
+
+    const ariaProps = {
+      isDisabled: disabled,
+      isReadOnly: readOnly,
+      isRequired: required,
+      ...rest
+    };
+
+    const state = useNumberFieldState({ ...ariaProps, locale: 'en-US' });
+    const inputRef = React.useRef<HTMLInputElement>(null);
+
+    React.useImperativeHandle(ref, () => inputRef.current as any);
+
+    const { inputProps } = useNumberField(ariaProps, state, inputRef);
 
     React.useEffect(() => {
-      if (!inputParent || !inputParent.current) return;
+      const handleWheel = (e: WheelEvent) => e.preventDefault();
```

Comment:
Is there a reason this has to be in the `useEffect`? The convention we use is to always declare functions outside the component body where possible (i.e. when a function isn't relying on component state).

Reply:
No need in this case - if it was shared in another component then yes, but in this case it's not used anywhere else (and it's a single line of code!) so it's more straightforward to declared it at the top of the file. If we did need it elsewhere, then we would move it out then.